controller/controller.go

```diff
@@ -584,84 +582,79 @@ func UpdateSchedulerConfig(
 
 	l.Info("starting scheduler update")
 
-	if maxSurge <= 0 {
-		return errors.New("invalid parameter: maxsurge must be greater than 0")
+	err := validateConfig(logger, configYAML, maxSurge)
+	if err != nil {
+		return err
 	}
 
-	if configYAML.AutoScaling.Max > 0 && configYAML.AutoScaling.Min > configYAML.AutoScaling.Max {
-		return errors.New("invalid parameter: autoscaling max must be greater than min")
-	}
+	// Lock watchers so they don't scale up or down and the scheduler is not
+	// overwritten with older version on database
+	// during the databse write phase
+	globalLockKey := models.GetSchedulerLockKey(config.GetString("watcher.lockKey"), configYAML.Name)
```

Comment:
Maybe using better names might help on confusion. `GetSchedulerLockKey` could mean any type of key. `GetSchedulerScalingLockKey` is a bit more obvious.